libsolidity/Compiler.cpp

```diff
@@ -118,34 +118,30 @@ set<FunctionDefinition const*> Compiler::getFunctionsNeededByConstructor(Functio
 
 void Compiler::appendFunctionSelector(ContractDefinition const& _contract)
 {
-	vector<FunctionDefinition const*> interfaceFunctions = _contract.getInterfaceFunctions();
+	map<FixedHash<4>, FunctionDefinition const*> interfaceFunctions = _contract.getInterfaceFunctions();
 	vector<eth::AssemblyItem> callDataUnpackerEntryPoints;
 
-	if (interfaceFunctions.size() > 255)
-		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("More than 255 public functions for contract."));
+	if (interfaceFunctions.size() > 4294967295) // 2 ** 32
```

Comment:
We probably already have a serious problem before that number, because the probability of hash collisions is almost 1 even for 2**31. Please check for hash collisions somewhere else (already at parse time, perhaps during type checking) and remove this check here.